nexus/reconfigurator/execution/src/dns.rs

```diff
@@ -271,13 +271,38 @@ pub fn blueprint_internal_dns_config(
             ) => (ServiceName::InternalNtp, address.port()),
             BlueprintZoneType::Clickhouse(
                 blueprint_zone_type::Clickhouse { address, .. },
-            ) => (ServiceName::Clickhouse, address.port()),
+            )
+            | BlueprintZoneType::ClickhouseServer(
+                blueprint_zone_type::ClickhouseServer { address, .. },
+            ) => {
+                // Add the HTTP and native TCP interfaces for ClickHouse data
+                // replicas. This adds the zone itself, so we need to continue
+                // back up to the loop over all the Omicron zones, rather than
+                // falling through to call `host_zone_with_one_backend()`.
+                let http_service = if matches!(
+                    &zone.zone_type,
+                    BlueprintZoneType::Clickhouse(_)
+                ) {
+                    ServiceName::Clickhouse
+                } else {
+                    ServiceName::ClickhouseServer
+                };
+
+                // Safety: This only fails if we provide the same zone id more
+                // than once, which should not be possible here.
+                dns_builder
+                    .host_zone_clickhouse(
```

Comment:
Do we need to make similar changes to RSS? It is still (unfortunately) an entirely-separate code path.

Reply:
Yes! I forgot about that, thank you for reminding me.

Reply:
Added in 262770f45